Meta/Lagom/ClangPlugins/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+find_package(Clang 16 CONFIG REQUIRED)
+add_custom_target(AllClangPlugins)
+
+function(clang_plugin target_name)
+    cmake_parse_arguments(CLANG_PLUGIN "" "" "SOURCES" ${ARGN})
+    add_library(${target_name} ${CLANG_PLUGIN_SOURCES})
+    install(TARGETS ${target_name} DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    target_include_directories(${target_name} SYSTEM PRIVATE ${CLANG_INCLUDE_DIRS} ${LLVM_INCLUDE_DIRS})
+    target_compile_features(${target_name} PRIVATE cxx_std_20)
+    target_compile_options(${target_name} PRIVATE -Wall -Wextra -Werror -Wno-unused -fno-rtti)
+
+    add_custom_target(${target_name}Target DEPENDS ${target_name})
+    add_dependencies(AllClangPlugins ${target_name}Target)
+
+    set_property(GLOBAL APPEND PROPERTY CLANG_PLUGINS_ALL_COMPILE_OPTIONS -fplugin=${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/lib${target_name}.so)
```

Comment:
where is this property used? or does the target-based solution in the following commits replace this idea?

Reply:
It's used in the `lit.site.cfg.py.in` template file. Is it possible to use the generator expressions for this? I feel like the answer is no, since it seems like they're evaluated at build time (not totally sure, I've never used them before), and the lit file is created at configure time. So either way I'll need to keep this ugly syntax.